Add Fargate deployment readme (#540)
signalfx/splunk-otel-collector: deployments/fargate/README.md

## Default Configuration
The default configuration file is located at `/etc/otel/collector/fargate_config.yaml`
in the Collector image. See 
[here](https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/config/collector/fargate_config.yaml)
for its contents and 
[here](https://github.com/signalfx/splunk-otel-collector/blob/main/cmd/otelcol/Dockerfile)
for the Collector image Dockerfile. Note that the default metrics receivers are `hostmetrics`
and `smartagent/ecs-metadata`.

Default configuration steps in the container definition for the Collector:
- Map the endpoint ports in the default configuration file (i.e. `13133`, `6060`,
  `55679`, `14250`, `6832`, `6831`, `14268`, `8888`, `7276`, `9943`, `9411`, `9080`).
- Assign the default configuration file path to environment variable `SPLUNK_CONFIG`.
- Assign a list of metrics you want excluded to environment variable `METRICS_TO_EXCLUDE`.
- Assign a list of images whose metrics you want excluded to environment variable `IMAGES_TO_EXCLUDE`.

## Custom Configuration
For the Collector pick up your custom configuration, you need to assign the path to your custom

> configuration file to environment variable `SPLUNK_CONFIG` in the container definition for

Reviewer: Isn't `SPLUNK_CONFIG_YAML` the only option to provide a custom config at the moment? With `SPLUNK_CONFIG` user can define only another existing pre-built config which are not custom but specific to other use-cases (agent, gateway). So we probably should put instructions from _Direct Configuration_ section here?

Author: No, `SPLUNK_CONFIG` can be used to specify non-default config file.

Reviewer: Sorry I don't understand. `SPLUNK_CONFIG` stores a file path to a config, right? If user cannot mount any external volume to the container, how can they provide a custom config?

Author: Users can mount volumes to tasks in Fargate. The mounted volume(s) are accessible to all containers in task. `SPLUNK_CONFIG_YAM` is just for convenience.

# AWS Fargate Deployment
Familiarity with AWS Fargate (Fargate) is assumed. Consult the 
[User Guide for AWS Fargate](https://docs.aws.amazon.com/AmazonECS/latest/userguide/what-is-fargate.html)
for further reading.

Unless stated otherwise, it is assumed that the

> [Splunk OpenTelemetry Collector](https://github.com/signalfx/splunk-otel-collector)

Reviewer 2: ```suggestion
[Splunk OpenTelemetry Connector](https://github.com/signalfx/splunk-otel-collector)
```

> (Collector) is deployed to an ECS Task in a **sidecar** container alongside monitored

Reviewer 2: ```suggestion
(Collector) is deployed as an ECS Task in a **sidecar** container alongside monitored
```

applications.

> Applies to Collector release v0.30.0 and above which corresponds to image tag 0.30.0 and above

Reviewer 2: ```suggestion
Requires Connector release v0.30.0 or newer which corresponds to image tag 0.30.0 and newer
```

in the image repository [here](https://quay.io/repository/signalfx/splunk-otel-collector?tab=tags).

> Overview of the container definition for the Collector:

Reviewer 2: It seems like this is a combination of 1) configuration options 2) explanation of what things are -- I would separate these things. In addition, I would put the getting started information as close to the top as possible.

Author: Ok

- Set environment variable `SPLUNK_CONFIG` to the configuration file path. Set `SPLUNK_CONFIG`

>   to the default configuration file path to enable default configuration.

Reviewer 2: Does it default to a setting and this is about overriding? It should in which case configuration options.

Author: Yes, the default configuration file for the collector container is /etc/otel/collector/gateway_config.yaml but we want the default in fargate to be /etc/otel/collector/fargate_config.yaml.

Reviewer 2: We support setting SPLUNK_CONFIG so getting started and data setup UI should include steps to ensure proper config is used.

Author: ok

> - Use extension `ecs_observer` in your custom configuration to discover targets.

Reviewer 2: The default config has this right? Is there a link to additional documentation? If not, why is this included?

Author: No, the default config does not have ecs_observer. Mainly because we don't know the task name or rather arn ahead of time.

- Enable ECS read-only permissions in the task role when using `ecs_observer`.
- Filter the targets for `ecs_observer` to be within task when the Collector is a sidecar,
  that is, the Collector and the monitored containers are in the same task.
- `ecs_observer` is currently limited to Prometheus targets.
- You can use environment variable `SPLUNK_CONFIG_YAML` instead of `SPLUNK_CONFIG` to specify
  your configuration YAML directly. No need for a file.

## Default Configuration

> The default configuration file is located at `/etc/otel/collector/fargate_config.yaml`

Reviewer 2: Great! Let's ensure getting started uses this by default

Author: Are you suggesting that `/etc/otel/collector/fargate_config.yaml` should be used by default without the need to set SPLUNK_CONFIG=`/etc/otel/collector/fargate_config.yaml`?

Reviewer 2: using the env var is fine -- stating that getting started steps and data setup UI steps should have this env var set properly.

Author: Included setting SPLUNK_CONFIG in getting started section

# AWS Fargate Deployment
Familiarity with AWS Fargate (Fargate) is assumed. Consult the 
[User Guide for AWS Fargate](https://docs.aws.amazon.com/AmazonECS/latest/userguide/what-is-fargate.html)
for further reading.

Unless stated otherwise, the
[Splunk OpenTelemetry Connector](https://github.com/signalfx/splunk-otel-collector)
(Collector) is deployed to an ECS task in a **sidecar** container. We are using the term
"sidecar" loosely here to mean having the Collector container as a sidecar to multiple or
all monitored application containers in the task. We do not mean a Collector sidecar container

> per monitored application container as in the strict sense.

Reviewer: This part seems a bit confusing. I don't understand why is it needed. Specifically this line `Collector container sidecar to multiple or all monitored application containers in the task` doesn't sound right to me.

The collector can be deployed as a sidecar -- another container to one existing task that is being monitored. Or is can be deployed as a standalone task -- like a gateway. I assume this document should specifically describe one way or another. In order to collect metrics from each task, I assume, it should be deployed as sidecar to each container. If so, we need to stick to that approach and document it as prescribed guideline.

Author: Ok, I removed the extra confusing bit.

Requires Connector release v0.30.0 or newer which corresponds to image tag 0.30.0 and newer.
See image repository [here](https://quay.io/repository/signalfx/splunk-otel-collector?tab=tags).

> ## Getting Started

Reviewer 2: Immediately following this header have a code block with something a user can copy and paste to get the connector working on Fargate.

Author: Added default container definition JSON for Collector.

The default configuration file is located at `/etc/otel/collector/fargate_config.yaml`
in the Collector image according to the Collector image Dockerfile
[here](../../cmd/otelcol/Dockerfile). See the contents of the default configuration file
[here](../../cmd/otelcol/config/collector/fargate_config.yaml). Note that receivers

> `hostmetrics` and `smartagent/ecs-metadata` are specified by default.

Reviewer: I think we should have a task definition example as provided in signalfx-agent https://github.com/signalfx/signalfx-agent/blob/main/deployments/fargate/example-fargate-task.json . Not necessary within this PR, probably can be another one.

Author: Added container definition JSON for the collector

Prometheus targets in cluster `lorem-ipsum-cluster`, region `us-west-2`, where the task arn
pattern is `^arn:aws:ecs:us-west-2:906383545488:task-definition/lorem-ipsum-task:[0-9]+$`.
The results are written to file `/etc/ecs_sd_targets.yaml`. The `prometheus` receiver is
configured to read targets from the results file. Note that the values for `access_token`
and `realm` are read from environment variables `SPLUNK_ACCESS_TOKEN` and `SPLUNK_REALM`
respectively, which must be specified in your container definition.

```yaml
extensions:
  ecs_observer:
    refresh_interval: 10s
    cluster_name: 'lorem-ipsum-cluster'
    cluster_region: 'us-west-2'
    result_file: '/etc/ecs_sd_targets.yaml'
    task_definitions:
      - arn_pattern: "^arn:aws:ecs:us-west-2:906383545488:task-definition/lorem-ipsum-task:[0-9]+$"
        metrics_ports: [9113]
        metrics_path: /metrics
receivers:
  signalfx:
  prometheus:
    config:
      scrape_configs:
        - job_name: 'lorem-ipsum-nginx'
          scrape_interval: 10s
          file_sd_configs:
            - files:
                - '/etc/ecs_sd_targets.yaml'
processors:
  batch:
  resourcedetection:

>     detectors: [ecs]

Reviewer 2: this doesn't have `system` yet fargate config does -- why?

Author: I'll remove the system detector from the default config

The example below shows an excerpt of the container definition JSON for the Collector 
configured to use custom configuration file `/path/to/custom/config/file`. 
`/path/to/custom/config/file` is a placeholder value for the actual custom configuration
file path and `0.32.0` is the latest image tag at present. The custom configuration file
should be present in a volume attached to the task.
```json
{
  "environment": [
    {
      "name": "SPLUNK_CONFIG",
      "value": "/path/to/custom/config/file"
    }
  ],
  "image": "quay.io/signalfx/splunk-otel-collector:0.32.0",
  "essential": true,
  "name": "splunk_otel_collector"
}
```
The custom Collector container definition essentially:
- Specifies the Collector image.
- Sets environment variable `SPLUNK_CONFIG` with the custom configuration file path.

Alternatively, you can specify the custom configuration YAML directly using environment
variable `SPLUNK_CONFIG_YAML` as describe [below](#direct-configuration).

### ecs_observer
Use extension
[Amazon Elastic Container Service Observer](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/ecsobserver#amazon-elastic-container-service-observer)

> (`ecsobserver` or `ecs_observer`) in your custom configuration to discover metrics targets

Reviewer 3: If you're referring to the component type I think only `ecs_observer` is valid.

```suggestion
(`ecs_observer`) in your custom configuration to discover metrics targets
```

(Collector) is deployed as a **sidecar** (additional container) to ECS tasks.

Requires Connector release v0.32.0 or newer which corresponds to image tag 0.32.0 and newer.
See image repository [here](https://quay.io/repository/signalfx/splunk-otel-collector?tab=tags).

## Getting Started
Copy the default Collector container definition JSON below. Replace `MY_SPLUNK_ACCESS_TOKEN`
and `MY_SPLUNK_REALM` with valid values. Update the image tag to the newest
version then add the JSON to the `containerDefinitions` section of your task definition
JSON.
```json
{
  "portMappings": [
    {
      "hostPort": 13133,
      "protocol": "tcp",
      "containerPort": 13133
    },
    {
      "hostPort": 6060,
      "protocol": "tcp",
      "containerPort": 6060
    },
    {
      "hostPort": 14250,
      "protocol": "tcp",
      "containerPort": 14250
    },
    {
      "hostPort": 9943,
      "protocol": "tcp",
      "containerPort": 9943
    },
    {
      "hostPort": 9411,
      "protocol": "tcp",
      "containerPort": 9411
    },
    {
      "hostPort": 9080,
      "protocol": "tcp",
      "containerPort": 9080
    }

>   ],

Reviewer: I don't think this is required. By default fargate network is shared between containers, so we should be able to send traces to localhost from other containers. I tested tracing, it works well

Author: Ok, I'll remove the port mappings.

  "environment": [
    {
      "name": "SPLUNK_ACCESS_TOKEN",
      "value": "MY_SPLUNK_ACCESS_TOKEN"
    },
    {
      "name": "SPLUNK_REALM",
      "value": "MY_SPLUNK_REALM"
    },
    {
      "name": "SPLUNK_CONFIG",

>       "value": "/etc/otel/collector/fargate_config.yaml"

Reviewer 3: Not needed for this PR, but it'd be nice if fargate envs were detected automatically so no specification is required.

Author: I don't know. We could have a dedicated image for Fargate in which the fargate config is picked up by default. Or we could have some flag for signaling the environment and pick the right default config. This would be not too much different from what we have now. I don't know. What do you have in mind?

Reviewer 3: I was referring to the default config selection logic: https://github.com/signalfx/splunk-otel-collector/blob/cba4c0aaf94a6895b5c19c6bac39954a1353284c/cmd/otelcol/main.go#L215

The example below shows an excerpt of the container definition JSON for the Collector 
configured to use custom configuration file `/path/to/custom/config/file`. 
`/path/to/custom/config/file` is a placeholder value for the actual custom configuration
file path and `0.31.0` is the latest image tag at present. The custome configuration file
should be present in a volume attached to the task.
```json
{
  "environment": [
    {
      "name": "SPLUNK_CONFIG",
      "value": "/path/to/custom/config/file"
    }
  ],
  "image": "quay.io/signalfx/splunk-otel-collector:0.31.0",
  "essential": true,
  "name": "splunk_otel_collector"
}
```
The custom Collector container definition essentially:
- Specifies the Collector image.
- Sets environment variable `SPLUNK_CONFIG` with the custom configuration file path.

Alternatively, you can specify the custom configuration YAML directly using environment
variable `SPLUNK_CONFIG_YAML` as describe [below](#direct-configuration).

### ecs_observer
Use extension
[Amazon Elastic Container Service Observer](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/ecsobserver#amazon-elastic-container-service-observer)

> (`ecsobserver` or `ecs_observer`) in your custom configuration to discover metrics targets

Reviewer 3: we don't include the ecs observer as a component do we?

# AWS Fargate Deployment
Familiarity with AWS Fargate (Fargate) is assumed. Consult the 
[User Guide for AWS Fargate](https://docs.aws.amazon.com/AmazonECS/latest/userguide/what-is-fargate.html) for further reading.

The
[Splunk OpenTelemetry Collector](https://github.com/signalfx/splunk-otel-collector)
(splunk-collector or collector) can be deployed to an ECS Task as a standalone container 

> or as sidecar container alongside the application containers that you want to monitor.

Reviewer 4: Suggested change: "... or as a sidecar container equal with the ..."

Author: I rewrote the entire readme

Find the Docker image repository for the collector

> [here](https://quay.io/repository/signalfx/splunk-otel-collector?tab=tags).

Reviewer 4: Suggested change: "See signalfx / splunk-otel-collector (https://quay.io/repository/signalfx/splunk-otel-collector?tab=tags) for the Docker image repository for the collector."

Author: I edited the readme entirely. Please see later commit.

> The latest tag is `quay.io/signalfx/splunk-otel-collector:latest`.

Author: Added a comment that changes apply to release >= v0.30.0

> ## Standalone Container

Reviewer 4: Suggested change: "To deploy AWS Fargate as a standalone container, add the container definition for the collector to a task that is separate from the task containing the application containers that you 
want to monitor. "

Author: I rewrote the entire readme

As a standalone container, you add the container definition for the collector to a
task that is separate from the task containing the application containers that you 
want to monitor. You configure the collector to use extension
[Amazon Elastic Container Service Observer (ecsobserver)](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/observer/ecsobserver#amazon-elastic-container-service-observer).
to discover targets in running tasks, filtered by service names, task definitions
and container labels. The ecsobserver is currently limited to discovering
prometheus targets. Also, the task role must include the read-only permissions below. 
The permissions can be added to a customer-managed policy that is attached to the
task role.

> 

Reviewer 4: Suggested change: "You can configure the collector to use the use the Amazon Elastic Container Service Observer (ecsobserver) extension to discover targets in running tasks, filtered by service names, task definitions, and container labels. The ecsobserver is currently limited to discovering Prometheus targets. The task role must also include the read-only permissions shown below. The permissions can be added to a customer-managed policy that is attached to the task role."

Author: I rewrote the entire readme

```text
ecs:List*
ecs:Describe*
```

> Below is an example of a collector configuration that configures extension

Reviewer 4: Suggested change: "The following is an example collector configuration showing the ecs_observer (ecsobserver) extension configured to to find Prometheus targets in the splunk-fargate-cluster cluster in the us-west-2 region with the NGINX_PROMETHEUS_EXPORTER_PORT container label. ecs_observer then writes the results to file /etc/ecs_sd_targets.yaml, which the Prometheus receiver is configured to read."

Author: I edited the readme entirely. Please see later commit.

> Below is an example of a collector configuration that configures extension

Author: I changed the example to filter by task definition arn

extensions:
  ecs_observer:
    refresh_interval: 10s
    cluster_name: 'splunk-fargate-cluster'
    cluster_region: 'us-west-2'
    result_file: '/etc/ecs_sd_targets.yaml'
    docker_labels:
      - port_label: 'NGINX_PROMETHEUS_EXPORTER_PORT'
receivers:
  signalfx:
  prometheus:
    config:
      scrape_configs:
        - job_name: 'splunk-fargate-standalone-nginx'
          scrape_interval: 10s
          file_sd_configs:
            - files:
                - '/etc/ecs_sd_targets.yaml'
processors:
  batch:
exporters:
  signalfx:
    access_token: ${SPLUNK_ACCESS_TOKEN}
    realm: ${SPLUNK_REALM}
service:
  extensions: [ecs_observer]
  pipelines:
    metrics:
      receivers: [prometheus]
      processors: [batch]
      exporters: [signalfx]
```

The configuration YAML can be supplied to the collector in a file whose path
is the value for flag `--config` or environment variable `SPLUNK_CONFIG`. The YAML
can also be specified directly at the commandline using environment variable 
`SPLUNK_CONFIG_YAML`. See the custom configuration section 
[here](https://github.com/signalfx/splunk-otel-collector/blob/main/docs/getting-started/linux-manual.md#custom-configuration)
for details. 

> 

Reviewer 4: Suggested change: "The YAML can also be specified directly at the command line using the SPLUNK_CONFIG_YAML environment variable. See custom configuration for more information."

Author: I rewrote the entire readme

In Fargate, access to the filesystem is not readily available so using
`SPLUNK_CONFIG_YAML` is the most convenient option. You can simply create a parameter
in the AWS Systems Manager (Amazon SSM) Parameter Store and have the configuration YAML as its 
value. Then in the container definition for the collector, you specify 
environment variable `SPLUNK_CONFIG_YAML` to get its value from the parameter.
Also, grant the collector access permissions to Amazon SSM by attaching policy
`AmazonSSMReadOnlyAccess` to the task role.

> 

Reviewer 4: Suggested change: "Use the SPLUNK_CONFIG_YAML environment variable to get access to the Fargate file system. You can create a parameter in the AWS Systems Manager (Amazon SSM) Parameter Store and have the configuration YAML as its value. Specify `SPLUNK_CONFIG_YAML` to get its value from the parameter in the container definition for the collector. Also, grant the collector access permissions to Amazon SSM by attaching the `AmazonSSMReadOnlyAccess` policy to the task role."

Are these things required? Do they need to be done in this order? If so, I would write this as a list of ordered steps.

Author: I added a list of steps as suggested.

Furthermore, note that the values for `access_token` and `realm` are provided via 
environment variables `SPLUNK_ACCESS_TOKEN` and `SPLUNK_REALM` so these environment
variables must be specified in the container definition.

> 

Reviewer 4: Suggested change: "Note also that the values for values for access_token and realm are provided by using the  SPLUNK_ACCESS_TOKEN and SPLUNK_REALM environment variables, which must be specified in the container definition."

Author: Ok, I changed the sentence as suggested

## Sidecar Container
For the sidecar container, the collector uses the
[Amazon ECS Resource Detector Processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md)
to query the
[Amazon ECS task metadata endpoint](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-fargate.html)

> for targets in the current task in which the collector container is running.

Reviewer 4: Suggested change: "To deploy AWS Fargate as a sidecar container, add the Amazon ECS Resource Detector Processor to your collector configuration. The Amazon ECS Resource Detector Processor queries the Amazon ECS task metadata endpoint for targets in the current task in which the collector container is running."

Reviewer 4: I would also consider adding a brief explanation of a standalone container vs. a sidecar container and why you would want to use one deployment mode over another deployment mode. Is there an environment where standalone container deployment is the preferred mode?

Author: I changed the readme to emphasize sidecar deployment. I now mention the "standalone" deployment as a by the way at the end of the readme. The agent is deployed as a sidecar and I just assume the collector will also be deployed as such. It's just at the ecsobserver in the collector gives it the capability to be deployed in a separate task and I think that's worth mentioning.

Unless stated otherwise, the
[Splunk OpenTelemetry Connector](https://github.com/signalfx/splunk-otel-collector)
(Collector) is deployed as a **sidecar** (additional container) to ECS tasks.

> Requires Connector release v0.31.0 or newer which corresponds to image tag 0.31.0 and newer.

Reviewer 2: This needs to be 32 right? (Requires another release post merge)

Author: Yes

See image repository [here](https://quay.io/repository/signalfx/splunk-otel-collector?tab=tags).

## Getting Started
Copy the default Collector container definition JSON below. Replace `MY_SPLUNK_ACCESS_TOKEN`
and `MY_SPLUNK_REALM` with valid values. Update the image tag to the newest
version then add the JSON to the `containerDefinitions` section of your task definition
JSON.
```json
{
  "portMappings": [
    {
      "hostPort": 13133,
      "protocol": "tcp",
      "containerPort": 13133
    },
    {
      "hostPort": 6060,
      "protocol": "tcp",
      "containerPort": 6060
    },
    {
      "hostPort": 14250,
      "protocol": "tcp",
      "containerPort": 14250
    },
    {
      "hostPort": 8888,
      "protocol": "tcp",
      "containerPort": 8888

>     },

Reviewer 2: This doesn't need to be made public, does it?

Author: No, it does not.

    {
      "hostPort": 7276,
      "protocol": "tcp",
      "containerPort": 7276

>     },

Reviewer 2: This is not needed as `sapm` receiver is removed right?

}
```
In the above container definition the Collector is configured to use the default
configuration file `/etc/otel/collector/fargate_config.yaml`. The Collector image Dockerfile
is available [here](../../cmd/otelcol/Dockerfile) and the contents of the default
configuration file can be seen [here](../../cmd/otelcol/config/collector/fargate_config.yaml).
Note that receivers `hostmetrics` and `smartagent/ecs-metadata` are specified by default.

In summary, the default Collector container definition:
- Specifies the Collector image.
- Maps the endpoint ports specified in the default configuration file.
- Sets environment variables `SPLUNK_ACCESS_TOKEN`, `SPLUNK_REALM` and `SPLUNK_CONFIG`.
- Optionally:
  - Assigns a list of metrics to exclude to environment variable `METRICS_TO_EXCLUDE`.
  - Assigns a list of images whose metrics should be excluded to environment variable `IMAGES_TO_EXCLUDE`.

## Custom Configuration
The example below shows an excerpt of the container definition JSON for the Collector 
configured to use custom configuration file `/path/to/custom/config/file`. 
`/path/to/custom/config/file` is a placeholder value for the actual custom configuration

> file path and `0.31.0` is the latest image tag at present. The custome configuration file

Reviewer 2: Should be 32

Author: Yes

should be present in a volume attached to the task.
```json
{
  "environment": [
    {
      "name": "SPLUNK_CONFIG",
      "value": "/path/to/custom/config/file"
    }
  ],

>   "image": "quay.io/signalfx/splunk-otel-collector:0.31.0",

Reviewer 2: 32

Author: Yes

      processors: [batch, resourcedetection]
      exporters: [signalfx]
```

> **Note:** The task arn pattern in the configuration example above will cause `ecs_observer`

Reviewer 2: ARN should be capitalized

Author: Ok

```

### Direct Configuration
In Fargate the filesystem is not readily available. This makes specifying the configuration
YAML directly instead of using a file more convenient. The Collector provides environment
variable `SPLUNK_CONFIG_YAML` for specifying the configuration YAML directly which can be
used instead of `SPLUNK_CONFIG`.

For example, you can store the custom configuration above in a parameter called
`splunk-otel-collector-config` in **AWS Systems Manager Parameter Store**. Then in your
Collector container definition assign the parameter to environment variable 
`SPLUNK_CONFIG_YAML` using `valueFrom`. The example below shows an excerpt of the container
definition JSON for the Collector. `MY_SPLUNK_ACCESS_TOKEN` and `MY_SPLUNK_REALM` are 

> placeholder values and image tag `0.31.0` is the latest at present.

Reviewer 2: 32

Author: Yes

```json
{
  "environment": [
    {
      "name": "SPLUNK_ACCESS_TOKEN",
      "value": "MY_SPLUNK_ACCESS_TOKEN"
    },
    {
      "name": "SPLUNK_REALM",
      "value": "MY_SPLUNK_REALM"
    }
  ],
  "secrets": [
    {
      "valueFrom": "splunk-otel-collector-config",
      "name": "SPLUNK_CONFIG_YAML"
    }
  ],

>   "image": "quay.io/signalfx/splunk-otel-collector:0.31.0",

Reviewer 2: 32

Author: Yes
